subsys/net/lib/dhcpv4/dhcpv4.c

```diff
@@ -1668,6 +1668,16 @@ static void dhcpv4_iface_event_handler(struct net_mgmt_event_callback *cb,
 			if (!net_if_ipv4_addr_rm(iface, &iface->config.dhcpv4.requested_ip)) {
 				NET_DBG("Failed to remove addr from iface");
 			}
+
+			/* Remove DNS servers as interface is gone. We only need to
+			 * do this for this interface. If using global setting, the
+			 * DNS servers are removed automatically when the interface
+			 * comes back up.
+			 */
+			if (IS_ENABLED(CONFIG_NET_DHCPV4_DNS_SERVER_VIA_INTERFACE)) {
```

Comment:
Shouldn't this only be done if `CONFIG_NET_DHCPV4_OPTION_DNS_ADDRESS` is enabled? I imagine otherwise DNS addresses could be statically configured.

Reply:
It is done only if `CONFIG_NET_DHCPV4_OPTION_DNS_ADDRESS` is enabled (see line 1120 above where this is enforced). Also there is a dependency in Kconfig for these options.

Reply:
Ah true, you're right, nevermind then.